WalletWasabi/Blockchain/BlockFilters/FilterProcessor.cs

```diff
@@ -64,7 +64,9 @@ public async Task ProcessAsync(uint serverBestHeight, FiltersResponseState filte
 				{
 					// Reorg happened
 					// 1. Rollback index
-					FilterModel reorgedFilter = await BitcoinStore.IndexStore.RemoveLastFilterAsync().ConfigureAwait(false);
+					FilterModel? reorgedFilter = await BitcoinStore.IndexStore.TryRemoveLastFilterAsync().ConfigureAwait(false)
+						?? throw new InvalidOperationException("Fatal error: Failed to remove the reorded filter.");
```

Comment:
Due to the null coalescing operator a resulting value is guaranteed to be not a null.

```suggestion
					FilterModel reorgedFilter = await BitcoinStore.IndexStore.TryRemoveLastFilterAsync().ConfigureAwait(false)
						?? throw new InvalidOperationException("Fatal error: Failed to remove the reorded filter.");
```